core/src/main/scala/org/bitcoins/core/protocol/blockchain/ChainParams.scala

```diff
@@ -143,23 +146,98 @@ sealed abstract class ChainParams {
     val genesisBlock = Block(genesisBlockHeader, Seq(tx))
     genesisBlock
   }
+
+  /**
+    * The minimum amount of proof of work required for a block
+    * [[https://github.com/bitcoin/bitcoin/blob/eb7daf4d600eeb631427c018a984a77a34aca66e/src/consensus/params.h#L70 bitcoin core pow limit]]
+    * @return
+    */
+  def powLimit: BigInteger
+
+  /** The targetted timespan between difficulty adjustments
+    * As of this implementation, all of these are the same in bitcoin core
+    *
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L73 mainnet]]
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L190 testnet]]
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L285 regtest]]
+    * */
+  def powTargetTimeSpan: Duration
+
+  /**
+    * Not sure what this value is right now, but it is constant across bitcoin
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L74 mainnet]]
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L191 testnet]]
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L286 regtest]]
+    * @return
+    */
+  def powTargetSpacing: Int
```

Comment:
Could this be how often we're supposed to have blocks? 10  * 60 seconds = 10 minutes